src/gflownet/models/mmc.py

```diff
@@ -0,0 +1,113 @@
+import numpy as np
+import hydra
+import wandb
+from omegaconf import OmegaConf
+
+from rdkit import Chem
+from rdkit.Chem import Draw
+from rdkit.Chem.rdchem import Mol as RDMol
+
+from typing import Optional
+from torch.nn import Module
+from pytorch_lightning import LightningModule
+
+from sklearn.metrics.pairwise import cosine_similarity
+import matplotlib.pyplot as plt
+
+from multimodal_contrastive.utils import utils
+from multimodal_contrastive.data.featurization import mol_to_data
+from multimodal_contrastive.networks.utils import move_batch_input_to_device
+
+OmegaConf.register_new_resolver("sum", lambda input_list: np.sum(input_list))
+
+
+class MMC_Proxy(Module):
+    model: Optional[Module]
+    latents: Optional[np.ndarray]
+    wrap_for_mp: bool = False
+
+    def __init__(self, cfg_name: str, cfg_dir: str, ckpt_path: str, worker: str):
+        super().__init__()
+        with hydra.initialize_config_dir(config_dir=cfg_dir, version_base=None):
+            cfg = hydra.compose(config_name=cfg_name)
+        model: LightningModule = utils.instantiate_model(cfg)
+        model = model.load_from_checkpoint(ckpt_path, map_location=worker)
+        model = model.to(worker)
+        self.model = model.eval()
+
+        # TODO make optional and load from arg path
+        self.latents = np.load("/home/mila/s/stephen.lu/gfn_gene/res/mmc/puma_embeddings.npz")
+
+    def log_target_properties(self, target, target_latent, mode="joint"):
+        fig, ax = plt.subplots(figsize=(5, 5))
+
+        # compute cosine similarity between target modality and all struct latents
+        cosine_sim = cosine_similarity(target_latent, self.latents["struct"])
```

Comment:
I am confused by this a bit (probably because I don't know what `self.latents` are - I assumed that they would be the latents of the targets in the dataset)?

Reply:
yeah sorry, I am plotting the cosine similarity between the target latent and the `struct` latents of all molecules in the training dataset to get a sense of how "difficult" it is to find similar representations.